src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -1532,6 +1532,13 @@ void InterpreterSelectQuery::addEmptySourceToQueryPlan(
     }
 }
 
+bool InterpreterSelectQuery::moveToPrewhereIfFinal()
```

Comment:
Let's rename to `shouldMoveToPrewhere` and also include the check `settings.optimize_move_to_prewhere` inside the function.